pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -299,47 +293,97 @@ func (s *Service) sendMsgEvents(msg *service.StateMsg) {
 func (s *Service) startInternalListener() {
 	go func() {
```

Comment:
I guess I missed this earlier - I would keep the go statement outside ( no nested go func() {} )